src/Core/src/Platform/Android/ContextExtensions.cs

```diff
@@ -104,15 +104,36 @@ public static float ToPixels(this Context? self, double dp)
 			return (float)Math.Ceiling(dp * s_displayDensity);
 		}
 
+		/// <summary>
+		/// Converts dp to pixels, rounding to the nearest whole pixel.
+		/// </summary>
+		/// <remarks>
+		/// This is especially useful when converting coordinates to the nearest pixel.
+		/// Do not use this method when setting sizes, as it may lead to visual truncation of the content
+		/// when rounding down.
+		/// </remarks>
+		internal static float ToPixelsRound(this Context? self, double dp)
+		{
+			EnsureMetrics(self);
+
+			return (float)Math.Round(dp * s_displayDensity, MidpointRounding.AwayFromZero);
```

Comment:
Apart from the UITest, could be nice to include some device tests checking conversions.